nimare/io.py

```diff
@@ -275,3 +278,123 @@ def convert_sleuth_to_dataset(text_file, target="ale_2mm"):
+        collection_ids can also be a dictionary whose keys are the informative
+        study name and the values are collection ids to give the collections
+        more informative names in the dataset.
+    contrasts : :obj:`dict`
+        Dictionary whose keys represent the name of the contrast in
+        the dataset and whose values represent how that contrast is identified
+        in neurovault.
+        For example, under the ``Name`` column in this URL
+        https://neurovault.org/collections/8836/,
+        a valid contrast could be "as-Animal", which will be called "animal" in the created
+        dataset if the contrasts argument is ``{'animal': ["as-Animal"]}``.
+    img_dir : :obj:`str` or None
+        Base path to save all the downloaded images, by default the images
+        will be saved to a temporary directory with the prefix "neurovault"
+    map_type_conversion : :obj:`dict` or None
+        Dictionary whose keys are what you expect the `map_type` name to
+        be in neurovault and the values are the name of the respective
+        statistic map in a nimare dataset. Default = None.
+    **dset_kwargs : keyword arguments passed to Dataset
+        Keyword arguments to pass in when creating the Dataset object.
+        see :obj:`nimare.dataset.Dataset` for details.
+
+    Returns
+    -------
+    :obj:`nimare.dataset.Dataset`
+        Dataset object containing experiment information from neurovault.
+    """
+    if img_dir is None:
+        img_dir = Path(tempfile.mkdtemp(prefix="neurovault_"))
+    else:
+        img_dir = Path(img_dir, exist_ok=True)
+        img_dir.mkdir(parents=True, exist_ok=True)
+
+    if map_type_conversion is None:
+        map_type_conversion = {
+            "T map": "t",
+            "variance": "varcope",
+            "univariate-beta map": "beta",
+        }
+
+    if not isinstance(collection_ids, dict):
+        collection_ids = {nv_coll: nv_coll for nv_coll in collection_ids}
+
+    dataset_dict = {}
+    for coll_name, nv_coll in collection_ids.items():
+        for contrast_name, contrast_ids in contrasts.items():
+            nv_url = f"https://neurovault.org/api/collections/{nv_coll}/images/?format=json"
+            images = requests.get(nv_url).json()
+
+            dataset_dict[f"study-{coll_name}"] = {
+                "contrasts": {
+                    contrast_name: {
+                        "images": {
+                            "beta": None,
+                            "t": None,
+                            "varcope": None,
+                        },
+                        "metadata": {"sample_sizes": None},
+                    }
+                }
+            }
+
+            for img_dict in images["results"]:
+                if not (
+                    (
+                        img_dict["name"] in contrast_ids
+                        or any(
+                            [
+                                img_dict["name"].startswith(f"{contrast}_stat")
```

Comment:
Not sure about the `_stat` part in the string; I think that will only work for BIDS-compliant maps. I'm not saying we should necessarily drop this, as it does seem a little bit liberal to just search for *any* image that starts with the contrast name. But maybe there's some intermediate approach that would make sense (or not—haven't thought about it).

Relatedly, is there any downside to allowing users to pass regexes as values in `contrast_ids`? That would dramatically increase flexibility (and probably obviate having to worry about the `_stat` thing above). To avoid having to add another argument that indicates whether or not to use strings as REs, we could check each value's type and only use `re.search` if it's a compiled regex instance. This is what we do in pybids and it seems to work pretty well.

Reply:
One other minor issue: I think checking `any()` here implies that any image in any collection that matches at least one of the `contrast_ids` values will be treated as a match. Is this what we want? I'm not saying it isn't, just that there's ambiguity right now about whether `contrast_ids` is a list of possible matches applied to every collection separately, or whether there's a one-to-one mapping from collections to entries in this list (as the example seems to suggest). I don't have a strong preference, but we should probably be consistent, and make sure it's clear in the docs.

Reply:
I believed I simplified the logic by allowing (forcing?) the user to pass a regex.